api/src/types/ast.ts

```diff
@@ -55,9 +55,26 @@ export type FunctionFieldNode = {
 	relatedCollection: string;
 };
 
+export type JsonFieldNode = {
+	type: 'jsonField';
+	// name of the database field
+	name: string;
+	// alias to use for the result
+	fieldKey: string;
+	// json-path to query
+	jsonPath: string;
+	// additional filters
+	query: Query;
+	// temporary nodes do not need to be in the final API response
+	// these are sometimes required for filtering
+	temporary: boolean;
+};
```

Comment:
Can this be part of the `FunctionNode`? To me, `json()` shouldn't be handled differently from other functions like `year()`. We should figure out how to upgrade FunctionNode to include optional arguments, that way we could also start implementing things like `multiply(<field>, 15)` over time